buildkite/bazelci.py

```diff
@@ -1285,6 +1304,54 @@ def execute_bazel_build(
         handle_bazel_failure(e, "build")
 
 
+def expand_test_target_patterns(bazel_binary, platform, test_targets):
+    if not any(t for t in test_targets if t.endswith(":all") or t.endswith("...")):
+        return test_targets
+
+    eprint("Resolving test targets via bazel query")
+    output = execute_command_and_get_output(
+        [bazel_binary]
+        + common_startup_flags(platform)
+        + [
+            "--nomaster_bazelrc",
+            "--bazelrc=/dev/null",
+            "query",
+            "tests(set({}))".format(" ".join("'{}'".format(t) for t in test_targets if t != "--")),
+        ],
+        print_output=False,
+    )
+    return [t for t in output.split("\n") if t.startswith("//")]
+
+
+def get_targets_for_shard(build_targets, test_targets, shard_id, shard_count):
+    # TODO(fweikert): implement a more sophisticated algorithm
+    all_targets = [("build", t) for t in sorted(build_targets)] + [
+        ("test", t) for t in sorted(test_targets)
+    ]
+    targets_for_this_shard = []
+    index = shard_id
+    target_count = len(all_targets)
+    while index < target_count:
+        targets_for_this_shard.append(all_targets[index])
+        index += shard_count
+
+    eprint(
```

Comment:
Can we remove this message? I think the information is visible in the resulting "bazel build ..." and "bazel test ..." lines anyway.